### PR TITLE
Ayzhuang/dex env

### PR DESCRIPTION
**Pull request by ayzhuang**

### CODE REVIEW

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 1]
```diff
@@ -1264,17 +1240,66 @@ void runtime::cpu::CPU_ExternalFunction::build()
...
+    // pass_manager.register_pass<runtime::cpu::pass::LSTMFusion>();
+    // pass_manager.register_pass<runtime::cpu::pass::RNNFusion>();
+    pass_manager.register_pass<ngraph::pass::AlgebraicSimplification>();
+    // pass_manager.register_pass<runtime::cpu::pass::MultiLayerRNNFusion>();
+    // pass_manager.register_pass<runtime::cpu::pass::ConcatInputs>();
+    pass_manager.register_pass<runtime::cpu::pass::CPUBatchFusion>();
+    pass_manager.register_pass<ngraph::pass::CommonSubexpressionElimination>();
+    pass_manager.register_pass<ngraph::pass::CoreFusion>();
+    pass_manager.register_pass<runtime::cpu::pass::CPUFusion>();
+    pass_manager.run_passes(m_function);
```
Comment (jmenon) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: This is incorrect for dex

Comment (jmenon) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: If it helps, I'd suggest auditing against a clean branch to make sure more stuff like this hasn't slipped in

Comment (jmenon) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: This is incorrect for dex.
If it helps, I'd suggest auditing against a clean branch to make sure more stuff like this hasn't slipped in

File: src/ngraph/runtime/cpu/cpu_external_function.cpp [hunk 2]
```diff
@@ -1026,14 +1014,6 @@ void runtime::cpu::CPU_ExternalFunction::build()
...
-    pass_manager.register_pass<runtime::cpu::pass::LSTMFusion>();
-    pass_manager.register_pass<runtime::cpu::pass::RNNFusion>();
-    pass_manager.register_pass<runtime::cpu::pass::ConcatInputs>();
-    pass_manager.register_pass<ngraph::pass::AlgebraicSimplification>();
-    pass_manager.register_pass<ngraph::pass::CommonSubexpressionElimination>();
-    pass_manager.register_pass<ngraph::pass::CoreFusion>();
-    pass_manager.register_pass<runtime::cpu::pass::CPUFusion>();
     pass_manager.register_pass<runtime::cpu::pass::CPUWorkspaceInsertion>(nv_cwi);
     pass_manager.register_pass<runtime::cpu::pass::CPUAssignment>(this);
     pass_manager.register_pass<runtime::cpu::pass::CPULayout>(this);
```
Comment (jbobba) [inline src/ngraph/runtime/cpu/cpu_external_function.cpp]: Cant we move these passes to register_and_run_passes?

Comment (ayzhuang) [reply src/ngraph/runtime/cpu/cpu_external_function.cpp]: You mean these three passes: CPUWorkspaceInsertion, CPUAssignment, and CPULayout? There are more passes after these three. compile() needs pass_manager. I tried to make pass_manager a class member, and it caused problem.

**Comment by jbobba** [comment]
@ayzhuang we can probably close this PR now that DEX is functionally complete.

**Pull request closed without merging**